pkg/test/framework/suite.go

```diff
@@ -247,6 +249,26 @@ func (s *suiteImpl) RequireEnvironmentVersion(version string) Suite {
 	return s
 }
 
+func (s *suiteImpl) RequireMaxVersion(version string) Suite {
+	fn := func(ctx resource.Context) error {
+		for _, c := range ctx.Clusters().Kube() {
+			ver, err := c.GetKubernetesVersion()
+			if err != nil {
+				return fmt.Errorf("failed to get Kubernetes version: %v", err)
+			}
+			serverVersion := fmt.Sprintf("%s.%s", ver.Major, ver.Minor)
+			if serverVersion > version {
```

Comment:
"if serverVersion > version" compares two strings based on the lexical order. Should the comparison be based on the numerical order instead? It is not straightforward to know the lexcial order of strings "1.12" and "11.2" (depending on the character order on the lexical table, "1.12" may be > or < "11.2") whereas the numerical order of 1.12 and 11.2 and is straightforward.

Reply:
You are correct but it also doesn't actually matter in practice since Kubernetes will never have a version 11.2. I can extract out this version stuff though since its causing some confusion

Reply:
I am working on an improvement to extract this logic out